src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -999,6 +999,130 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             return this;
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+
+            if (value != null && value.Length > 0)
+            {
+                if (value != this)
```

Comment:
`ToString` operates on the same instance of `StringBuilder` where `Append` would take 2 distinct instances.

Stepping back -- why did you have to special case the `Append` when being called with `this`? I think the same reasons apply if `Append` is called with another instance of `StringBuilder`.